### PR TITLE
Create a code fix that offers to add a reference to System.ValueTuple if it is missing and you use a tuple.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Note: i was able to get this working by using reflection to call into the appropriate nuget installer APIs.

**Comment by CyrusNajmabadi** [comment]
Tagging @srivatsn @Pilchie @jcouv

**Comment by CyrusNajmabadi** [comment]
test windows_debug_unit64_prtest  please

File: src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_NuGet.cs [hunk 1]
```diff
@@ -60,7 +60,7 @@ protected override IList<CodeAction> MassageActions(IList<CodeAction> actions)
                 var installerServiceMock = new Mock<IPackageInstallerService>(MockBehavior.Loose);
                 installerServiceMock.SetupGet(i => i.IsEnabled).Returns(true);
                 installerServiceMock.SetupGet(i => i.PackageSources).Returns(NugetPackageSources);
-                installerServiceMock.Setup(s => s.TryInstallPackage(It.IsAny<Workspace>(), It.IsAny<DocumentId>(), It.IsAny<string>(), "NuGetPackage", It.IsAny<string>(), It.IsAny<CancellationToken>()))
+                installerServiceMock.Setup(s => s.TryInstallPackage(It.IsAny<Workspace>(), It.IsAny<DocumentId>(), It.IsAny<string>(), "NuGetPackage", It.IsAny<string>(), It.IsAny<bool>(), It.IsAny<CancellationToken>()))
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_NuGet.cs]: You complain about other people making you scroll in code reviews...

File: src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs [hunk 2]
```diff
@@ -588,6 +602,20 @@ public void InstallPackage(string source, EnvDTE.Project project, string package
                 _packageInstaller.InstallPackage(source, project, packageId, version, ignoreDependencies);
             }
 
+            public void InstallLatestPackage(string source, EnvDTE.Project project, string packageId, bool includePrerelease, bool ignoreDependencies)
+            {
+                // Use reflection until the 3.5.0 version of NuGet.VisualStudio is released
+                // publicly.  Then we call into this method directly.
+                var installLatestPackageInfo = _packageInstaller.GetType().GetMethod(
```
Comment (Pilchie) [inline src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs]: Tracking bug for this?

Also - this is always public, just not in the nuget package?  Or we should have some different `BindingFlags`?

File: src/Workspaces/Core/Desktop/SymbolSearch/SymbolSearchUpdateEngine.cs [hunk 3]
```diff
@@ -109,11 +109,28 @@ public SymbolSearchUpdateEngine(ISymbolSearchLogService logService)
...
-            if (!_sourceToDatabase.TryGetValue(source, out databaseWrapper))
+            var result = await FindPackagesWithAssemblyWorkerAsync(source, assemblyName).ConfigureAwait(false);
+
+#if DEBUG
+            // For testing purposes, we hardcode this in if we're in DEBUG mode.
+            if (result.Length == 0 &&
+                source == NugetOrgSource &&
+                assemblyName == "System.ValueTuple")
+            {
+                return ImmutableArray.Create(new PackageWithAssemblyResult("System.ValueTuple", "", rank: 0));
```
Comment (Pilchie) [inline src/Workspaces/Core/Desktop/SymbolSearch/SymbolSearchUpdateEngine.cs]: I don't like this at all :(

Comment (Pilchie) [inline src/Workspaces/Core/Desktop/SymbolSearch/SymbolSearchUpdateEngine.cs]: Is this just because the package isn't in the index yet?

Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Desktop/SymbolSearch/SymbolSearchUpdateEngine.cs]: Yes.  @jinujoseph Is going to whitelist this for our index.

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit32_prtest  please

**Comment by Pilchie** [comment]
Can we retarget this to `dev15-rc2` and get it merged?  Are we still waiting on an update to the index?

**Comment by CyrusNajmabadi** [comment]
Yes.  @jinujoseph Is working on the Index side of things.   I'll see about retargeting dev-rc2.

**Comment by Pilchie** [comment]
@CyrusNajmabadi Can you add the ask mode template info?

Tagging @MattGertz for approval for RC.2.

**Comment by MattGertz** [comment]
@Pilchie @CyrusNajmabadi We'll also need another signoff before I can approve, in addition to the template.

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
test this please

**Pull request closed without merging**